pkg/bootstrapper/service.go

```diff
@@ -0,0 +1,258 @@
+package bootstrapper
+
+import (
+	"fmt"
+	"time"
+
+	"golang.org/x/sys/windows/svc"
+	"golang.org/x/sys/windows/svc/mgr"
+)
+
+// kubeletService struct contains the kubelet specific service information
+type kubeletService struct {
+	// obj is a pointer to the Windows service object
+	obj *mgr.Service
+	// svcDependents contains a list of services dependent on the current service
+	svcDependents []*mgr.Service
+}
+
+// newKubeletService creates and returns a new kubeletService object
+func newKubeletService(svcMgr *mgr.Mgr, ksvc *mgr.Service) (*kubeletService, error) {
+	var svcDependents []*mgr.Service
+
+	if svcMgr == nil || ksvc == nil {
+		return nil, fmt.Errorf("service manager or service object should not be nil")
+	}
+	if dependentSvc, err := svcMgr.OpenService(kubeletDependentSvc); err == nil {
+		svcDependents = append(svcDependents, dependentSvc)
+	}
+	return &kubeletService{
+		obj:           ksvc,
+		svcDependents: svcDependents,
+	}, nil
+}
+
+// getObj returns the service object for kubelet Windows service
+func (service *kubeletService) getObj() *mgr.Service {
+	if service.obj == nil {
+		return nil
+	}
+	return service.obj
+}
+
+// getConfig retrieves service config from service object Config()
+func (service *kubeletService) getConfig() (mgr.Config, error) {
+	config, err := service.obj.Config()
+	if err != nil {
+		return mgr.Config{}, err
+	}
+	return config, nil
+}
+
+// startKubeletService starts the kubelet as a Windows service
+func (service *kubeletService) startKubeletService() error {
+	if service.obj == nil {
+		return fmt.Errorf("no kubelet service found")
+	}
+	// if the service is running we do not need to start it
+	isServiceRunning, err := service.isKubeletServiceRunning()
+	if err != nil {
+		return fmt.Errorf("unable to check if Windows service is running: %v", err)
+	}
+	if isServiceRunning {
+		return nil
+	}
+	if err := service.obj.Start(); err != nil {
+		return err
+	}
+
+	if len(service.svcDependents) != 0 {
+		for _, dependent := range service.svcDependents {
+			err := startDependentService(dependent)
+			if err != nil {
+				return fmt.Errorf("unable to start dependent service for %s", KubeletServiceName)
+			}
+		}
+	}
+	return nil
+}
+
+// startDependentService starts the kubelet dependent Windows service
+func startDependentService(serviceObj *mgr.Service) error {
```

Comment:
Does this function actually start the dependants, or does it just start the service of whats passed into it?
If the latter, this should be renamed to startService. A function should be named according to what it does, not how its intended to be used